### PR TITLE
Migrate to signac-flow 0.5

### PR DESCRIPTION
**Pull request by csadorf**
While we claim that with the release of signac-flow 0.5 there is no need for a template anymore, it turns out that having a template is still really useful for documentation purposes and possibly even for project bootstrapping.

This pull request is the migration of the template for signac-flow version 0.5.x.
These are the major updates:

 * A lot of boiler plate code has been removed.
 * All source code is in `src/`.
 * The project is no longer implemented as part of a python package, but just in a module.
 * The template documentation in `doc/` has been removed, users should rather refer to the signac-flow documentation directly.
 * The template is compatible with Python 2.7 and testing for 2.7 is enabled.
 * Tests for Python version 3.4 have been disabled, since conda-forge does not distribute for 3.4 anymore. The signac-flow package is still tested for 3.4, just not distributed through conda-forge for that particular version, which would make testing here cumbersome.

### CODE REVIEW

File: src/project.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        self.add_operation(
+            name='sample',
+            cmd='python src/operations.py sample {job._id}',
+            pre=[self.initialized],
+            post=[self.sampled])
+
+    # Overload functions for execution script generation (optional)
+
+    def write_script_header(self, script, walltime=None, **kwargs):
+        super(MyProject, self).write_script_header(script, **kwargs)
```
Comment (csadorf) [inline src/project.py]: Missing in super method call: `walltime=walltime, `

File: src/project.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from flow import FlowProject
+from flow import staticlabel
+# import flow.environments  # uncomment to use default environments
+
+# import environments # uncomment to use custom environments
+
+import logging
+
+
+logger = logging.getLogger(__name__)
```
Comment (csadorf) [inline src/project.py]: I don't think the logger is used, should be removed.

**Comment by vyasr** [comment]
This looks fine to me. I think we could also achieve some of this clarity by making the default project.py generated by flow show at least a few of the more common methods that users may want to override (just having them call the corresponding super() methods) such as eligible, write_script, etc. For now this seems appropriate as a bridge while we figure that piece out.

**Review by vyasr — APPROVED** [review]

**Comment by csadorf** [comment]
@vyasr I agree, although we already have more elaborate examples when calling the `flow init` command with one of the other templates, e.g., `--template=example`.

I think what would be better than maintaining a single project template with a lot of boiler plate, is to have a library of different templates that clearly demonstrate how to solve different "problems". I have implemented the same kind of projects many times now and I've seen other people do the same, again solving the same problems. A library would allow us to demonstrate and distribute specialized solutions while keeping what is needed for the *base* FlowProject implementation as lean as possible.

I imagine a library that has GROMACS templates and HOOMD templates for instance. 
We could even add the option to add template source URLs or whatever so that when you initialize a project, it is automatically cloned from there.

**Comment by vyasr** [comment]
I like the idea of having many templates. That would definitely solve a lot of the problems. The one point I would make is that most of these templates (like the ones you're suggesting) would effectively provide more examples on the operations side. On the cluster submission side, though, I see how the current version can be quite opaque even for users not looking to make any changes, which is why I advocated for including more of that directly. I'm not advocating making things more heavyweight, just exposing some of what's happening in a more transparent manner.

**Comment by csadorf** [comment]
I've created a proposal for this on bitbucket: https://bitbucket.org/glotzer/signac-flow/issues/9/proposal-a-library-for-project-templates

**Comment by csadorf** [comment]
This pull request is obsolete.

**Pull request closed without merging**